src/query/ast/src/parser/token.rs

```diff
@@ -125,8 +125,8 @@ pub enum TokenKind {
     #[regex(r"[0-9]+")]
     LiteralInteger,
 
-    #[regex(r"[0-9]+e[+-]?[0-9]+")]
-    #[regex(r"([0-9]*\.[0-9]+(e[+-]?[0-9]+)?)|([0-9]+\.[0-9]*(e[+-]?[0-9]+)?)")]
+    #[regex(r"[0-9]+[eE][+-]?[0-9]+")]
+    #[regex(r"([0-9]*\.[0-9]+([eE][+-]?[0-9]+)?)|([0-9]+\.[0-9]*([eE][+-]?[0-9]+)?)")]
```

Comment:
Is is a good idea to use https://docs.rs/nom/latest/nom/number/complete/fn.float.html directly?

Reply:
use `nom::number::complete::float` to parse `LiteralFloat` token?

Reply:
How to use it into `derive`?

Reply:
But other Literals are implemented with regular expressions, should they be replaced by `nom`?

Reply:
cc @andylokandy

Reply:
tokenizer usually use regular expression instead of parser, because the formaer is faster and must terminate. nom is a parser, so we can't use it here.